allensdk/brain_observatory/behavior/session_apis/data_io/behavior_ophys_lims_api.py

```diff
@@ -0,0 +1,164 @@
+import logging
+from typing import Optional
+import pandas as pd
+
+from allensdk.api.cache import memoize
+from allensdk.brain_observatory.behavior.session_apis.data_io.ophys_lims_api \
+    import OphysLimsApi
+from allensdk.brain_observatory.behavior.session_apis.data_io import (
+    BehaviorLimsApi)
+from allensdk.internal.api import db_connection_creator, PostgresQueryMixin
+from allensdk.internal.core.lims_utilities import safe_system_path
+from allensdk.core.auth_config import (
+    LIMS_DB_CREDENTIAL_MAP, MTRAIN_DB_CREDENTIAL_MAP)
+from allensdk.core.authentication import credential_injector, DbCredentials
+from allensdk.brain_observatory.behavior.session_apis.data_transforms import (
+    BehaviorOphysDataXforms)
+
+
+class BehaviorOphysLimsApi(BehaviorOphysDataXforms,  OphysLimsApi,
+                           BehaviorLimsApi):
+
+    def __init__(self, ophys_experiment_id: int,
+                 lims_credentials: Optional[DbCredentials] = None,
+                 mtrain_credentials: Optional[DbCredentials] = None):
+
+        self.logger = logging.getLogger(self.__class__.__name__)
+
+        self.lims_db = db_connection_creator(
+            credentials=lims_credentials,
+            default_credentials=LIMS_DB_CREDENTIAL_MAP)
+
+        self.mtrain_db = db_connection_creator(
+            credentials=mtrain_credentials,
+            default_credentials=MTRAIN_DB_CREDENTIAL_MAP)
+
+        self.ophys_experiment_id = ophys_experiment_id
+        self.behavior_session_id = self.get_behavior_session_id()
+
+    def get_ophys_experiment_id(self):
+        return self.ophys_experiment_id
+
+    @memoize
+    def get_ophys_session_id(self):
+        query = '''
+                SELECT os.id FROM ophys_sessions os
+                JOIN ophys_experiment oe ON oe.ophys_session_id = os.id
+                WHERE oe.id = {};
+                '''.format(self.get_ophys_experiment_id())
+        return self.lims_db.fetchone(query, strict=True)
+
+    @memoize
+    def get_experiment_container_id(self):
+        query = '''
+                SELECT visual_behavior_experiment_container_id
+                FROM ophys_experiments_visual_behavior_experiment_containers
+                WHERE ophys_experiment_id= {};
+                '''.format(self.get_ophys_experiment_id())
+        return self.lims_db.fetchone(query, strict=False)
+
+    @memoize
+    def get_behavior_stimulus_file(self):
+        query = '''
+                SELECT stim.storage_directory || stim.filename AS stim_file
+                FROM ophys_experiments oe
+                JOIN ophys_sessions os ON oe.ophys_session_id = os.id
+                JOIN behavior_sessions bs ON bs.ophys_session_id=os.id
+                LEFT JOIN well_known_files stim ON stim.attachable_id=bs.id AND stim.attachable_type = 'BehaviorSession' AND stim.well_known_file_type_id IN (SELECT id FROM well_known_file_types WHERE name = 'StimulusPickle')
```

Comment:
can add some carriage returns before `AND`?

Reply:
Done!